app/uk/gov/hmrc/hec/services/FileCreationService.scala

```diff
@@ -37,87 +42,169 @@ trait FileCreationService {
+    case PAYE          => TaxSituationMapping(Some('N'), Some('Y'), None)
+    case SA            => TaxSituationMapping(Some('Y'), Some('N'), None)
+    case SAPAYE        => TaxSituationMapping(Some('Y'), Some('Y'), None)
+    case NotChargeable => TaxSituationMapping(None, None, Some('Y'))
+  }
+
+  case class SAStatusMapping(returnReceived: Option[Char], noticeToFileIssued: Option[Char])
+
+  private def saStatusMapping(saStatus: Option[SAStatus]) = saStatus match {
+    case Some(ReturnFound)        => SAStatusMapping(Some('Y'), None)
+    case Some(NoticeToFileIssued) => SAStatusMapping(Some('N'), Some('Y'))
+    case Some(NoReturnFound)      => SAStatusMapping(Some('N'), Some('N'))
+    case None                     => SAStatusMapping(None, None)
   }
+
+  private def createHecTaxCheckFileBody(hecTaxCheckList: List[HECTaxCheck]): List[HECTaxCheckFileBody] =
+    hecTaxCheckList.map { hecTaxCheck =>
+      hecTaxCheck.taxCheckData match {
+        case i: IndividualHECTaxCheckData =>
+          HECTaxCheckFileBody(
+            ggCredID = Some(i.applicantDetails.ggCredId.value),
+            nino = Some(i.taxDetails.nino.value),
+            firstName = Some(i.applicantDetails.name.firstName),
+            lastName = Some(i.applicantDetails.name.lastName),
+            dob = Some(i.applicantDetails.dateOfBirth.value.format(DATE_FORMATTER)),
+            SAUTR = i.taxDetails.sautr.map(_.value.toInt),
+            licenceType = licenceTypeEKV(i.licenceDetails.licenceType)._1,
+            licenceValidityPeriod = licenceValidityPeriodEKV(i.licenceDetails.licenceValidityPeriod)._1,
+            licenceTimeTrading = licenceTimeTradingEKV(i.licenceDetails.licenceTimeTrading)._1,
+            entityType = 'I',
+            notChargeable = taxSituationMapping(i.taxDetails.taxSituation).NotChargeable,
+            PAYE = taxSituationMapping(i.taxDetails.taxSituation).PAYE,
+            SA = taxSituationMapping(i.taxDetails.taxSituation).SA,
+            incomeTaxYear = i.taxDetails.saStatusResponse.map(_.taxYear.startYear + 1),
+            returnReceived = saStatusMapping(i.taxDetails.saStatusResponse.map(_.status)).returnReceived,
+            noticeToFile = saStatusMapping(i.taxDetails.saStatusResponse.map(_.status)).noticeToFileIssued,
+            taxComplianceDeclaration = saStatusMapping(i.taxDetails.saStatusResponse.map(_.status)).returnReceived,
+            customerDeclaration = 'Y',
+            taxCheckStartDateTime =
+              i.taxCheckStartDateTime.withZoneSameInstant(ZoneId.of("GMT")).format(DATE_TIME_FORMATTER),
+            taxCheckCompleteDateTime =
+              hecTaxCheck.createDate.withZoneSameInstant(ZoneId.of("GMT")).format(DATE_TIME_FORMATTER),
+            taxCheckCode = hecTaxCheck.taxCheckCode.value,
+            taxCheckExpiryDate = hecTaxCheck.expiresAfter.format(DATE_FORMATTER),
+            onlineApplication = if (i.source == Digital) 'Y' else 'N'
```

Comment:
does wart remover not complain about using `==` here?

Reply:
nopes 🤷‍♀️ , i can edit it